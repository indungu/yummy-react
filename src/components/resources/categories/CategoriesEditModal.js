import React from 'react';

export default (props) => {
  return (
    <div className="modal" id={`edit${props.name}`} tabIndex="-1" role="dialog" style={{ display: props.display }}>
      <div className="modal-dialog" role="document">
        <div className="modal-content">
          <div className="modal-header">
            <h5 className="modal-title" id="categoryModalLabel">Edit Recipe Category</h5>
            <button type="button" className="close" onClick={props.hide}>
              <span aria-hidden="true">Ã—</span>
            </button>
          </div>
          <form onSubmit={props.onEditSubmit}>
            <div className="modal-body">
              <div className="form-group">
                <dt>
                  <label className="text-primary text-regular" htmlFor="categoryName">Name</label>
                </dt>
                <dd>
                  <input
                    className="form-control"
                    name="categoryName"
                    value={props.name}
                    onChange={props.onChange}
                  />
                </dd>
                <dt>
                  <label className="text-primary text-regular" htmlFor="categoryDescription">Description</label>
                </dt>
                <dd>
                  <textarea
                    className="form-control"
                    name="categoryDescription"
                    value={props.description}
                    onChange={props.onChange}
                  >
                  </textarea>
                </dd>
              </div>
            </div>
            <div className="modal-footer">
              <button type="button" className="btn btn-secondary" onClick={props.hide}>Close</button>
              <button
                type="submit"
                className="btn btn-success"
                onClick={props.onEditSubmit}
              >Update Category
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};
