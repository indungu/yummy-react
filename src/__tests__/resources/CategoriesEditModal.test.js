import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import CategoriesEditModal from '../../components/resources/categories/CategoriesEditModal';

describe('CategoriesEditModal', () => {
  const wrapper = shallow(<CategoriesEditModal />);
  it('Renders properly ', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
