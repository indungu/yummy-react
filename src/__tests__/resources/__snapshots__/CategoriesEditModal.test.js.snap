// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesEditModal Renders properly  1`] = `
<div
  className="modal"
  id="editundefined"
  role="dialog"
  style={
    Object {
      "display": undefined,
    }
  }
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="categoryModalLabel"
        >
          Edit Recipe Category
        </h5>
        <button
          className="close"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <form>
        <div
          className="modal-body"
        >
          <div
            className="form-group"
          >
            <dt>
              <label
                className="text-primary text-regular"
                htmlFor="categoryName"
              >
                Name
              </label>
            </dt>
            <dd>
              <input
                className="form-control"
                name="categoryName"
              />
            </dd>
            <dt>
              <label
                className="text-primary text-regular"
                htmlFor="categoryDescription"
              >
                Description
              </label>
            </dt>
            <dd>
              <textarea
                className="form-control"
                name="categoryDescription"
              />
            </dd>
          </div>
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-secondary"
            type="button"
          >
            Close
          </button>
          <button
            className="btn btn-success"
            type="submit"
          >
            Update Category
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
